README.md

```diff
@@ -19,113 +19,122 @@ This library includes the following modules.
   - [`Group[T]`](https://typelevel.org/cats/api/cats/kernel/Group.html)
-- `magnolify-scalacheck` - type class derivation for [ScalaCheck](https://github.com/typelevel/scalacheck) 
+- `magnolify-scalacheck` - type class derivation for [ScalaCheck](https://github.com/typelevel/scalacheck)
   - [`Arbitrary[T]`](https://github.com/typelevel/scalacheck/blob/master/doc/UserGuide.md#universally-quantified-properties)
-  - [`Cogen[T]`](https://github.com/typelevel/scalacheck/blob/master/src/main/scala/org/scalacheck/Cogen.scala#L20-L29)
+  - [`Cogen[T]`](https://github.com/typelevel/scalacheck/blob/master/src/main/scala/org/scalacheck/Cogen.scala)
 - `magnolify-guava` - type class derivation for [Guava](https://guava.dev)
-  - [`Funnel[T]`](https://guava.dev/releases/25.0-jre/api/docs/com/google/common/hash/Funnel.html)
-  
-- `magnolify-avro` - conversion between Scala case classes and [Apache Avro](https://github.com/apache/avro) `GenericRecord`
-- `magnolify-bigquery` - conversion between Scala case classes and [Google Cloud BigQuery](https://cloud.google.com/bigquery/) `TableRow`
-- `magnolify-datastore` - conversion between Scala case classes and [Google Cloud Datastore](https://cloud.google.com/datastore/) `Entity`
-- `magnolify-tensorflow` - conversion between Scala case classes and [TensorFlow](https://www.tensorflow.org/) `Example`
+  - [`Funnel[T]`](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/hash/Funnel.html)
+
+- `magnolify-avro` - conversion between Scala types and [Apache Avro](https://github.com/apache/avro) `GenericRecord`
+- `magnolify-bigquery` - conversion between Scala types and [Google Cloud BigQuery](https://cloud.google.com/bigquery/) `TableRow`
+- `magnolify-datastore` - conversion between Scala types and [Google Cloud Datastore](https://cloud.google.com/datastore/) `Entity`
+- `magnolify-tensorflow` - conversion between Scala types and [TensorFlow](https://www.tensorflow.org/) `Example`
 
 # Usage
 
 Cats, ScalaCheck, and Guava type class derivation can be performed both automatically and semi-automatically.
 
-Automatic derivation with implicits:
+Automatic derivation works out of the box by importing implicits from `magnolify.$module.auto._`.
+It works with context bounds i.e. `def plus[T: Semigroup](x: T, y: T)` and implicit paramemters i.e.
+`def f[T](x: T, y: T)(implicit sg: Semigroup[T])`.
+
 ```scala
-case class MyCaseClass(field: NestedCaseClass) 
-case class NestedCaseClass(innerField: String) // works with other non-String elements, this is just an example
+case class Inner(int: Int, str: String)
+case class Outer(inner: Inner)
 
-// implicitly keyword summons an instance of the typeclass wrapping a given case class
-// typeclasses are generated at compile time with macros & Magnolia
-// this works if each element within the case class has a typeclass instance defined within the implicit scope
+// implicitly[T] summons an implicit instance of T
```

Comment:
What this does vs what's perceived to be happening.
Also IMO typeclasses don't wrap anything, they just attach new behavior in an ad-hoc way.

Reply:
Actually I'm not sure we should even explain what `implicitly` does here. It's used only to summon the implicit and is not supposed be used by end user. The goal of README is to show quickly how something works and this is irrelevant.